tools/nuget/validate_package.py

```diff
@@ -117,24 +118,33 @@ def check_if_dlls_are_present(
 
             if package_type == "nuget":
                 folder = "runtimes/" + platform + "/" + native_folder
-                header_folder = "build/native/include"
+                build_dir = "buildTransitive" if is_gpu_package else "build"
+                header_folder = f"{build_dir}/native/include"
             else:  # zip package
                 folder = package_path + "/lib"
                 header_folder = package_path + "/include"
 
-            path = folder + "/" + "onnxruntime.dll"
-            print("Checking path: " + path)
-            if path not in file_list_in_package:
-                print("onnxruntime.dll not found for " + platform)
-                raise Exception("onnxruntime.dll not found for " + platform)
-
-            if is_gpu_package:
+            # In Nuget GPU package, onnxruntime.dll is in dependent package.
+            if not (package_type == "nuget" and is_gpu_package):
+                path = folder + "/" + "onnxruntime.dll"
+                print("Checking path: " + path)
+                if path not in file_list_in_package:
+                    print("onnxruntime.dll not found for " + platform)
+                    raise Exception("onnxruntime.dll not found for " + platform)
+                continue
+
+            # In Nuget GPU package, cuda ep and tensorrt ep dlls are in dependent package.
+            if (is_gpu_package and package_type != "nuget") or is_gpu_dependent_package:
                 for dll in win_gpu_package_libraries:
                     path = folder + "/" + dll
                     print("Checking path: " + path)
                     if path not in file_list_in_package:
                         print(dll + " not found for " + platform)
                         raise Exception(dll + " not found for " + platform)
+            # In Nuget GPU package, gpu header files are not in dependent package.
+            if (is_gpu_package and package_type != "nuget") or (
+                package_type == "nuget" and not is_gpu_dependent_package
+            ):
```

Comment:
This logic is a little hard to follow. Would it be simpler if we set a couple of variables and check for the library/headers based on that? 

e.g. 
```py
package_contains_library = False if package_type == "nuget" and is_gpu_package else True
package_contains_headers = False if is_gpu_dependent_package else True
```

Would be good to check the package does not contain headers as well when that is the expected state.

Reply:
thx. updated